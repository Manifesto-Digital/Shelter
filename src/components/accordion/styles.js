import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
