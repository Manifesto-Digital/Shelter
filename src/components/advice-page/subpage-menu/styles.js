import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
