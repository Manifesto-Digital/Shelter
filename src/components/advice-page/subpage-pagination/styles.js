import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
