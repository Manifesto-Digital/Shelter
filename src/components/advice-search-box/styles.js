import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
