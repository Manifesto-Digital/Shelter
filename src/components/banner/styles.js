import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
