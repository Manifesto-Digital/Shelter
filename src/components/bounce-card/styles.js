import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
