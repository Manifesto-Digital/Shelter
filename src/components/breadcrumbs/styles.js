import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
