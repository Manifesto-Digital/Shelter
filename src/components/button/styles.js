import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonStyles } from '../styled/buttons';

export const DefaultButton = styled.button`
  ${buttonStyles};
`;

export const IconHolder = styled(SVG)`
  fill: ${({ iconColour, theme }) =>
    (iconColour === 'black' && theme.palette.black) || theme.palette.white};

  display: inline-block;
  width: 20px;
  height: 20px;
  margin-right: 10px;
  vertical-align: top;
  margin-top: -2px;
`;
