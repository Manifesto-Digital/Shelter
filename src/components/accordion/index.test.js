import React from 'react';
import { act } from 'react-dom/test-utils';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import Accordion from '.';
import { HeadingButton, Content } from './styles';
import { createFactory } from '../../utils/test-factories';

export const createAccordion = createFactory({
  id: 'ID123',
  active: true,
  header: 'mock header',
  children: (
    <p>
      Phasellus volutpat, metus eget egestas mollis, lacus lacus blandit dui, id
      egestas quam mauris ut lacus.
    </p>
  ),
});

it('renders correctly', () => {
  const mockData = createAccordion();
  snapshotComponent(<Accordion {...mockData} />);
});

it('is hidden and shows the correct aria attributes', () => {
  const mockData = createAccordion({ id: 'mockID' });
  const wrapper = mountWithTheme(<Accordion {...mockData} />);

  expect(wrapper.find(Content)).toHaveStyleRule('display', 'none');
  expect(wrapper.find(HeadingButton).prop('aria-controls')).toEqual(
    mockData.id
  );
  expect(wrapper.find(HeadingButton).prop('aria-expanded')).toEqual(false);
  expect(wrapper.find(Content).prop('aria-hidden')).toEqual(true);
});

it('opens and updates aria on click', () => {
  const mockData = createAccordion();
  const wrapper = mountWithTheme(<Accordion {...mockData} />);

  act(() => {
    wrapper.find(HeadingButton).simulate('click');
  });
  wrapper.update();
  expect(wrapper.find(Content)).toHaveStyleRule('display', 'block');
  expect(wrapper.find(HeadingButton).prop('aria-controls')).toEqual(
    mockData.id
  );
  expect(wrapper.find(HeadingButton).prop('aria-expanded')).toEqual(true);
  expect(wrapper.find(Content).prop('aria-hidden')).toEqual(false);
});
