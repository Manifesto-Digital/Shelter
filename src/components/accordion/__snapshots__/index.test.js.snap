// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  border-bottom: 1px solid #e5e5e5;
  margin-bottom: 20px;
}

.c1 {
  position: relative;
  font-weight: normal;
  color: #e5e5e5;
  margin-bottom: 0;
}

.c2 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  padding: 20px 60px 20px 5px;
  width: 100%;
  text-align: left;
  font-size: 1.35em;
}

.c4 {
  display: none;
  padding: 20px 5px 0 5px;
}

.c3 {
  position: absolute;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  top: 50%;
  right: 5px;
  width: 20px;
  height: 25px;
}

.c3 svg {
  display: block;
  width: 20px;
  height: 25px;
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    <button
      aria-controls="ID123"
      aria-expanded={false}
      className="c2"
      onClick={[Function]}
      type="button"
    >
      mock header
    </button>
    <span
      className="isvg loading c3"
    />
  </h3>
  <div
    aria-hidden={true}
    className="c4"
    id="ID123"
  >
    <p>
      Phasellus volutpat, metus eget egestas mollis, lacus lacus blandit dui, id egestas quam mauris ut lacus.
    </p>
  </div>
</div>
`;
